backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/ConflictGenerator.java

```diff
@@ -151,7 +151,7 @@ private static AutogenVersionEnvironmentDiff getVersionDiff(
     Map<String, AutogenPythonRequirementEnvironmentBlob> mapC = getRequirementMap(c);
     HashSet<String> keys = new HashSet<>();
     keys.addAll(mapA.keySet());
-    keys.addAll(mapB.keySet());
+    keys.retainAll(mapB.keySet());
```

Comment:
Is this doing intersections instead of union?

Reply:
yes